harmony_netcdf_to_zarr/adapter.py

```diff
@@ -85,7 +85,7 @@ def process_items_many_to_one(self):
         """
         workdir = mkdtemp()
         try:
-            items = list(self.catalog.get_items())
+            items = list(self.get_all_catalog_items(self.catalog))
```

Comment:
The `harmony-service-lib-py` dependency in `requirements/core.txt` should probably be updated to the version of the library that contains `BaseHarmonyAdapter.get_all_catalog_items`. Also looking at that file, it might be cleaner to just grab `harmony-service-lib-py` from PyPI in the standard way, too (e.g., `harmony-service-lib ~= 1.0.16`).

I think the way the current dependency is specified will still pull the right version, but it feels tidier to specify the minimum version we can run with.